### PR TITLE
Chromaprint (fpcalc)

### PR DESCRIPTION
**Pull request by cytec**
discussion: https://github.com/SynoCommunity/spksrc/issues/902
test spk: https://cytec.us/syno/test-spk/chromaprint/

### CODE REVIEW

**Comment by rednoah** [comment]
In that case /usr/local/bin should work. Just need to keep in mind that /usr/local/bin doesn't exist by default. I'm pretty sure it's not affected by DSM update like /usr/syno/bin is. 

I'd be careful with /usr/syno/bin and /usr/bin because those folders will most likely be reset whenever DSM is updated.

File: cross/ffmpeg-2.2.1/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+HOMEPAGE = http://www.ffmpeg.org/
+COMMENT  = FFmpeg is a complete, cross-platform solution to record, convert and stream audio and video. It includes libavcodec - the leading audio/video codec library
+LICENSE  =
+
+REQUIRED_DSM = 5.1
+
+# Define SYNCTHING_ARCH as per SyncThing standards
+ifeq ($(findstring $(ARCH),88f6281),$(ARCH))
+MYARCH = armv5
```
Comment (Dr-Bean) [inline cross/ffmpeg-2.2.1/Makefile]: This caught my eye (didn't check the rest of the PR too closely yet). 
I could be mistaken, but my earlier investigation showed that ffmpeg doesn't care at all about which arm arch you're trying to compile, just that it contains `arm`. The reason for the `FFMPEG_ARCH` patches was that the word `armada` is not recognized further on in the compile process, or something like that (I forgot already...)
If that's correct, you could just lump all the arm-based arches together and do `FFMPEG_ARCH=arm`.

File: spk/chromaprint/src/installer.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+preinst ()
+{
+    exit 0
+}
+
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+    if [ ! -e "$TARGET_LINK" ]; then
```
Comment (Dr-Bean) [inline spk/chromaprint/src/installer.sh]: You probably want an `mkdir -p /usr/local/bin` in here too.

File: spk/chromaprint/src/installer.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    rm -f ${TARGET_LINK}
+
+    exit 0
+}
+
+postuninst ()
+{
+    # Remove link
+    rm -f ${INSTALL_DIR}
```
Comment (Dr-Bean) [inline spk/chromaprint/src/installer.sh]: ...and remove the symlink during uninstall.

File: cross/ffmpeg-2.2.1/Makefile (hunk 1 above)
Comment (cytec) [inline cross/ffmpeg-2.2.1/Makefile]: yup if i remember correctly i've got an error like `armada is not a known arch` or something like that, will putt everything in `FFMPEG_ARCH=arm`

**Comment by Diaoul** [comment]
It's hard to see which PR is under work and which PR should be reviewed. I like [docker labels](https://github.com/docker/docker/issues) maybe we can do the same approache (simplified ofc)?

**Comment by cytec** [comment]
@Diaoul i'm 100% with you on that one. i like especially the status labels which make clear what's going on:

    status/0-needs-triage
    status/1-needs-design-review
    status/2-needs-code-review
    status/3-needs-docs-review
    status/4-ready-to-merge
    status/assigned
    status/claimed
    status/closed/cannot-reproduce
    status/closed/duplicate
    status/closed/invalid
    status/docs-help
    status/help-wanted
    status/more-info-needed
    status/needs-escalation

and use them with those we already have?

**Comment by cytec** [comment]
@Diaoul i've just got some time and looked through those labels we already have and wanted to assign them to a few issues (at least some of them i'm currently working at) ... but seems like i'm not allowed to do that based on my permissions :D

**Comment by Diaoul** [comment]
That's weird, you tried here: https://github.com/SynoCommunity/spksrc/labels ?

**Comment by cytec** [comment]
@Diaoul yup. can't add new ones and can't assign existing ones to issues/PR's :/

**Comment by Dr-Bean** [comment]
I believe only repository members can create and assign labels. Makes sense, e.g. from a triaging point of view, you don't want people to assign their own labels, right?

**Comment by Diaoul** [comment]
I changed the permission please try again

**Comment by cytec** [comment]
@Diaoul still the same for me... but seems like @Dr-Bean changed from `Collaborator` to `Owner`

**Comment by Dr-Bean** [comment]
That can't be right, right? ;) 
Anyway, it seem you need write access at minimum to set labels, if [this](https://help.github.com/articles/permission-levels-for-an-organization-repository) is still up to date. I think that's what Diaoul is setting up?

**Comment by cytec** [comment]
yeah :D now it's working. thanks

**Comment by cytec** [comment]
@Diaoul @Dr-Bean i've some new labels and assigned a few of them to some of the recent issues... maybe we'll create a new issue to discuss this further? ex: what labels do we need, when should which label be assigned and what color should they have etc.

**Comment by Diaoul** [comment]
@cytec awesome thanks :+1:

**Comment by Diaoul** [comment]
LGTM

File: cross/chromaprint/Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+PKG_NAME = chromaprint
+PKG_VERS = 1.2
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://bitbucket.org/acoustid/chromaprint/downloads
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/ffmpeg-2.2.1
```
Comment (Dr-Bean) [inline cross/chromaprint/Makefile]: Can you check if using `BUILD_DEPENDS` here works for ffmpeg (BUILD_DEPENDS skips PLIST processing)? 
If so, you don't have to clean up the stuff in the postinstall here: https://github.com/SynoCommunity/spksrc/pull/1622/files#diff-b18021950851bbb4e94acd6843c4156cR37.

File: cross/ffmpeg-2.2.1/Makefile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+DEPENDS = cross/x264 cross/lame
+
+HOMEPAGE = http://www.ffmpeg.org/
+COMMENT  = FFmpeg is a complete, cross-platform solution to record, convert and stream audio and video. It includes libavcodec - the leading audio/video codec library
+LICENSE  =
+
+REQUIRED_DSM = 5.1
+
+# Define SYNCTHING_ARCH as per SyncThing standards
```
Comment (Dr-Bean) [inline cross/ffmpeg-2.2.1/Makefile]: Replace the Syncthing references :)

**Comment by cytec** [comment]
@Dr-Bean `BUILD_DEPEND` seemed to work, thanks :)

File: cross/ffmpeg-2.2.1/Makefile [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+PKG_DIST_SITE = http://www.ffmpeg.org/releases/
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/x264 cross/lame
+
+HOMEPAGE = http://www.ffmpeg.org/
+COMMENT  = FFmpeg is a complete, cross-platform solution to record, convert and stream audio and video. It includes libavcodec - the leading audio/video codec library
+LICENSE  =
+
+REQUIRED_DSM = 5.1
```
Comment (Dr-Bean) [inline cross/ffmpeg-2.2.1/Makefile]: @cytec Noticed this while compiling, is there an issue with older toolchains? We don't do this with the 'other' ffmpeg.

**Comment by Dr-Bean** [comment]
Chromaprint 1.2-1 has been published on the repository on the beta channel. Thanks @cytec

**Pull request merged**